### ISSUE TITLE
[Form] New component

### ISSUE THREAD

**Issue by kylegach** [mineral-ui/mineral-ui#693]
As a user of an app built with Mineral UI, I would like a nice way to see all of the errors in a given set of FormFields, with well-considered a11y, so that my experience is a good one.

**Comment by kylegach** [comment]
Helpful info from the a11y Slack:

> Hi all, right now I'm doing a refactor of all the forms in my company's app and I'm looking for some form validation advice. It's the subjective, more UX leaning decisions that keep tripping me up. I'm creating an 'error summary' above the form. A couple questions:
> 
> 1. Should I send focus to the error summary container div, or to the heading within it? 
> 2. Should the error summary be an aria-live element and an alert, or is it enough that focus is sent there?
> 3. I've seen a pattern where each error in the summary is a link that sends focus to that specific input. Is that advisable/preferred/etc?
> 4. If you do an error summary like this, should you also do inline errors, or is that redundant?

------

> 1. Heading within should be fine
> 2. Sending focus is enough. doing both would create redundant announcements
> 3. This can be quite nice for users, as they can go directly to the form control that has the error, rather than having to tab through other focus stops to get there.
> 4. Inline errors are still important to help indicate which form controls have the error, so that you aren't relying on color alone.  Depending on screen size, the summary may be out of view, so these inline errors will help provide context in that situation as well.
